src/main/java/app/gpx_animator/core/renderer/cache/TileCache.java

```diff
@@ -78,32 +78,36 @@ public static void ageCache() {
         }
     }
 
-    public static BufferedImage getTile(final String url, final String tileCacheDir, final Long tileCacheTimeLimit) throws UserException {
+    public static BufferedImage getTile(final String url, final String userAgent, final String tileCacheDir, final Long tileCacheTimeLimit)
+        throws UserException {
 
         BufferedImage image;
 
         if (cachingEnabled(tileCacheDir)) {
             try {
-                image = cachedGetTile(url, tileCacheDir, tileCacheTimeLimit);
+                image = cachedGetTile(url, userAgent, tileCacheDir, tileCacheTimeLimit);
             } catch (final UserException e) {
-                image = unCachedGetTile(url);
+                image = unCachedGetTile(url, userAgent);
             }
         } else {
-            image = unCachedGetTile(url);
+            image = unCachedGetTile(url, userAgent);
         }
         return image;
     }
 
-    private static BufferedImage unCachedGetTile(final String url) throws UserException {
+    private static BufferedImage unCachedGetTile(final String url, final String userAgent) throws UserException {
         BufferedImage mapTile;
 
-        final var userAgent = String.format("%s %s on %s %s (%s)", //NON-NLS
-                Constants.APPNAME, Constants.VERSION, Constants.OS_NAME, Constants.OS_VERSION, Constants.OS_ARCH);
-        System.setProperty("http.agent", userAgent);
+        if (userAgent != null) {
```

Comment:
In `Configuration.java` you set the user agent to an empty string. To be safe, you should do a `userAgent.isBlank()` check here, too.

Reply:
Fixed